src/subscribe-substrate/lib/postgres.ts

```diff
@@ -595,11 +601,60 @@ export const deleteCommentActivityWithActivityStream = async (userId: string,com
         LEFT JOIN df.account_followers ON df.activities.account = df.account_followers.following_comment_id WHERE comment_id = $2)
     RETURNING *`
   const hexCommentId = encodeStructId(commentId);
-  const params = [userId, hexCommentId];
+  const params = [ userId, hexCommentId ];
   try {
     const res = await pool.query(query, params)
     console.log(res.rows)
   } catch (err) {
     console.log(err.stack);
   }
 }
+
+export const updateUnreadNotifications = async (account: string, activityId: number) => {
+  const defaultUnreadCount = 1;
+  const query = `
+    INSERT INTO df.notifications_counter (account, last_read_activity_id, unread_count)
+    VALUES ($1, $2, $3)
+    ON CONFLICT (account) DO UPDATE
+    SET unread_count =
+        (SELECT DISTINCT COUNT(*)
+          FROM df.activities
+          WHERE id IN ( 
+            SELECT activity_id
+            FROM df.notifications
+            WHERE account = $1
+            AND  activity_id >=
+              (SELECT last_read_activity_id FROM df.notifications_counter
+                WHERE account = $1
+              )
+            ) 
+            AND aggregated = true
+        )
+        last_read_activity_id = $2
+  `
+
+  const currentUnreadCount = await getUnreadNotifications(account) || 0
+  const params = [ account, activityId, defaultUnreadCount ]
+  try {
+    const res = await pool.query(query, params)
```

Comment:
Why we call `await getUnreadNotifications()` before `await pool.query(query, params)`?
In this way we get old value for unread notifications, don't we?

Reply:
true